tests/checkpoint/test_checkpoint.py

```diff
@@ -1112,3 +1268,283 @@ def test_newstyle_checkpoint_config_substitution_simple(
+            },
+            runtime_configuration={
+                "result_format": "BASIC",
+                "partial_unexpected_count": 999,
+                "template_1_key": 123,
+                "template_3_key": "bloopy!",
+                "new_runtime_config_key": "bleepy!",
+            },
+        )
+    )
+
+    substituted_config_template_and_runtime_kwargs = nested_checkpoint.get_substituted_config(
+        runtime_kwargs={
+            "expectation_suite_name": "runtime_suite_name",
+            "template_name": "my_nested_checkpoint_template_3",
+            "validations": [
+                {
+                    "batch_request": {
+                        "datasource_name": "my_datasource",
+                        "data_connector_name": "my_other_data_connector_2_runtime",
+                        "data_asset_name": "users",
+                        "partition_request": {"partition_index": -3},
+                    }
+                },
+                {
+                    "batch_request": {
+                        "datasource_name": "my_datasource",
+                        "data_connector_name": "my_other_data_connector_3_runtime",
+                        "data_asset_name": "users",
+                        "partition_request": {"partition_index": -4},
+                    }
+                },
+            ],
+            "run_name_template": "runtime_run_template",
+            "action_list": [
+                {
+                    "name": "store_validation_result",
+                    "action": {
+                        "class_name": "StoreValidationResultAction",
+                    },
+                },
+                {
+                    "name": "store_evaluation_params",
+                    "action": {
+                        "class_name": "MyCustomRuntimeStoreEvaluationParametersAction",
+                    },
+                },
+                {
+                    "name": "update_data_docs",
+                    "action": None,
+                },
+                {
+                    "name": "update_data_docs_deluxe_runtime",
+                    "action": {
+                        "class_name": "UpdateDataDocsAction",
+                    },
+                },
+            ],
+            "evaluation_parameters": {
+                "environment": "runtime-$GE_ENVIRONMENT",
+                "tolerance": 1.0e-2,
+                "aux_param_0": "runtime-$MY_PARAM",
+                "aux_param_1": "1 + $MY_PARAM",
+                "new_runtime_eval_param": "bloopy!",
+            },
+            "runtime_configuration": {
+                "result_format": "BASIC",
+                "partial_unexpected_count": 999,
+                "new_runtime_config_key": "bleepy!",
+            },
+        }
+    )
+    assert (
+        substituted_config_template_and_runtime_kwargs.to_json_dict()
+        == expected_nested_checkpoint_config_template_and_runtime_template_name.to_json_dict()
+    )
```

Comment:
Same comment as above -- could we reuse the tests I created by simply uncommenting out the `$MY_VAR` type statements and removing comments with my name in them?..

Reply:
Probably - agree that we should avoid duplication.